docs/ca/dev/monitoring/opentelemetry-instrumentation.md

```diff
@@ -0,0 +1,440 @@
+
+
+
+### Wire a console command
+
+Spryker is a big application, so you definitely don't want to create all hooks manually. Console command  class *OpentelemetryGeneratorConsole*  is used to generate those hooks automatically for classes that you want to cover with spans.
+```php
+<?php
+
+namespace Pyz\Zed\Console;
+
+...
+use Spryker\Zed\Opentelemetry\Communication\Plugin\Console\OpentelemetryGeneratorConsole;
+...
+
+class ConsoleDependencyProvider extends SprykerConsoleDependencyProvider
+{
+    protected function getConsoleCommands(Container $container): array
+    {
+        $commands = [
+            ...
+            new OpentelemetryGeneratorConsole(),
+            ...
+        ];
+
+        return $commands;
+    }
+}
+
+```
+
+You have to wire this console command into your install script, so it's executed on every deployment during base container image build. We recommend placing it as the last one in the section **build**.
+Please make sure that this command is located after all code modifications and generation.
+
+```yaml
+sections:
+    build:
+        generate-open-telemetry:
+            command: 'vendor/bin/console open-telemetry:generate'
+```
+
+### Configure hooks generation
+
+`open-telemetry:generate` command will do a heavy lifting and generate hooks for your code base. You can control what exactly you want to instrument by updating a few configuration methods.
+
+`\Spryker\Zed\Opentelemetry\OpentelemetryConfig::getExcludedDirs()` - controls what directories MUST NOT be instrumented. You may not want to see in your traces spans from some infra code that is not relevant for you. So you can just exclude the whole directory with a provided name. OOTB we excluded a bunch of directories. Review them in the vendor dir of the module if you want to instrument something from that list.
+Below you can see an example of how this method should look.
+```php
+class OpentelemetryConfig extends AbstractBundleConfig
+{
+    // We don't want to have traces for Monitoring module, OpenTelemetry module and tests in existing module, so those will be excluded even if we scan those directories during the generation. In the actual implementation you can see more directories.
+    public function getExcludedDirs(): array
+    {
+        return [
+        ...
+            'Monitoring',
+            'OpenTelemetry',
+            'tests',
+        ...
+        ];
+    }
+}
+```
+
+`\Spryker\Zed\Opentelemetry\OpentelemetryConfig::getExcludedSpans()` - allows you to exclude specific spans by name. You can use it to exclude autogenerated span that is not relevant for you, but excluding the directory itself is too much.
```

Comment:
Why is it `too much`?

Reply:
By excluding directory you can exclude other spans that you still want to see in the traces